crates/optimism/primitives/src/transaction/signed.rs

```diff
@@ -47,14 +61,9 @@ pub struct OpTransactionSigned {
 }
 
 impl OpTransactionSigned {
-    /// Calculates hash of given transaction and signature and returns new instance.
-    pub fn new(transaction: OpTypedTransaction, signature: Signature) -> Self {
-        let signed_tx = Self::new_unhashed(transaction, signature);
-        if signed_tx.ty() != OpTxType::Deposit {
-            signed_tx.hash.get_or_init(|| signed_tx.recalculate_hash());
-        }
-
-        signed_tx
```

Comment:
this doesn't have to be removed? you can rename it `new_unhashed`

Reply:
this mirrors eths txsigned type, we already have new_unhashed, so this change is ok